ouroboros-consensus-cardano-test/src/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs

```diff
@@ -130,11 +132,17 @@ type ShelleyBasedHardForkConstraints era1 era2 =
   , SL.TranslationContext era2 ~ ()
   )
 
-instance ShelleyBasedHardForkConstraints era1 era2
+instance ( ShelleyBasedHardForkConstraints era1 era2
+         , State (LC.EraRule "PPUP" era1) ~ SL.PPUPState era1
+         , State (LC.EraRule "PPUP" era2) ~ SL.PPUPState era2
+         )
```

Comment:
The `StyleGuide.md` wants

```
instance ( Eq a
         , ..
         ) => C a where
```

That seems to be consistent with other instances already in this file, too. See for example

https://github.com/input-output-hk/ouroboros-network/blob/07b8d4ca6e22960d627411e1466a498110086d1e/ouroboros-consensus-cardano-test/src/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs#L275